### ISSUE TITLE
SwiftUI support

### ISSUE THREAD

**Issue by Xerosigma** [icerockdev/moko-mvvm#36]
If supported - Could you provide an example of how binding would work using **SwiftUI**?

**Comment by Alex009** [comment]
Hi! in one of our forks already tried integrate with SwiftUI - https://github.com/icerockdev/moko-mvvm/compare/master...jacobminer:jm/swift-ui-support maybe it work and help you
we still not try SwiftUI self, but will try later and update implementation where needed

**Comment by michaelbrainvu** [comment]
The above integration did not work for me since the BindableObject can not be found so compilation failed.
Eventually I had to write a SwiftUI ObservableObject wrapper class which internally creates the actual view model and registers an event dispatcher listener. The model properties are mapped to using @Published properties which are set by actual view model properties observers. 
Notice that you can not use moko-units for generating list items unless you want to wrap UITableView for SwiftUI which misses the whole point of using SwiftUI.
See attached example
[ExampleViewModelWrapper.swift.zip](https://github.com/icerockdev/moko-mvvm/files/4711865/ExampleViewModelWrapper.swift.zip)

**Comment by jacobminer** [comment]
As far as I can tell, it's not possible for Moko-MVVM to support SwiftUI directly in the shared code right now because Kotlin Multiplatform **doesn't support interoperability with pure Swift modules** (like SwiftUI and Combine). 
In order to support SwiftUI, there are two main changes that need to occur, both of which are blocked by the above issue:
- Making the iOS `ViewModel` implementation conform to SwiftUI's `ObservableObject` protocol
- Making `LiveData` include an instance of Combine's `ObservableObjectPublisher` which we call `.send()` on when the internal value changes.

If the above were true, I _think_ we would be able to use SwiftUI as follows:

ViewModel
```kotlin
class SwiftUIViewModel: ViewModel() {
    val textLiveData = MutableLiveData("Hello world")
}
```

SwiftUI
```swift
struct SwiftUIView: View {
    @ObservedObject var viewModel: SwiftUIViewModel

    var body: some View {
        Text(viewModel.textLiveData.value)
    }
}
```

**Comment by Alex009** [comment]
thanks for your explanation. we will try to improve support of SwiftUI, maybe with additional swift cocoapod, what will have all required swift interop and objc conversion to use from kotlin-multiplatform side

**Comment by ln-12** [comment]
How's the progress in this? Do you plan to support it in the near future? As I want to develop an iOS app using SwiftUI, I need to decide if I could use this library.

**Comment by Alex009** [comment]
we not use SwiftUI now, so we not add support in nearby plans. but you can try add support self. if needed some consultations - feel free to ask here

**Comment by skajake** [comment]
+1 for SwiftUI support

**Comment by deviant-studio** [comment]
any updates on this? huge blocker for us :(

**Comment by Alex009** [comment]
sample will be available in 0.13.0 in sample-declarative-ui directory
also read 0.13.0 release notes and readme about dependency to SwiftUI swift additions
